src/main/scala/mesosphere/marathon/core/storage/simple/AsyncCuratorBuilderFactory.scala

```diff
@@ -0,0 +1,261 @@
+package mesosphere.marathon
+package core.storage.simple
+
+import org.apache.curator.framework.CuratorFramework
+import org.apache.curator.framework.imps.CuratorFrameworkState
+import org.apache.curator.x.async.AsyncCuratorFramework
+import org.apache.curator.x.async.api._
+import org.apache.zookeeper.CreateMode
+import org.apache.zookeeper.data.ACL
+
+import scala.collection.JavaConverters
+
+/**
+  * Simple case class holding factory's configuration options. Sensible defaults are provided for all options.
+  *
+  * @param createOptions a set with create options. By default node parents are created if needed and node data is compressed
+  * @param deleteOptions a set with delete options. By default node children are deleted if the node doesn't exist the delete method will appear to succeed
+  * @param existsOptions a set with exists options. By default no options are set
+  * @param createMode node create mode. By default [[CreateMode.PERSISTENT]] nodes are created
+  * @param acl node ACL. By default an empty list is used
+  * @param nodeVersion node version used to update node data or delete it. By default, -1 is used which matches all versions
+  * @param compressedData true if node data compression is enabled. Compression is enabled by default
+  */
+case class AsyncCuratorBuilderSettings(
+    createOptions: Set[CreateOption] = Set(CreateOption.createParentsIfNeeded, CreateOption.compress),
+    deleteOptions: Set[DeleteOption] = Set(DeleteOption.deletingChildrenIfNeeded, DeleteOption.quietly),
+    existsOptions: Set[ExistsOption] = Set.empty,
+    createMode: CreateMode = CreateMode.PERSISTENT,
+    acl: Seq[ACL] = Seq.empty,
+    nodeVersion: Int = -1,
+    compressedData: Boolean = true)
+
+/**
+  * A simple Factory that give an instance of the [[CuratorFramework]] provides
+  * factory methods to return async builders e.g. [[AsyncCreateBuilder]], [[AsyncDeleteBuilder]] etc.
+  *
+  * @param curator an instance of the [[CuratorFramework]]
+  * @param defaults async builder settings which are used as default parameters which simplifies request building. However
+  *             each factory method also allows overriding the defaults for every call.
+  */
+class AsyncCuratorBuilderFactory(curator: CuratorFramework, defaults: AsyncCuratorBuilderSettings) {
+
+  val async: AsyncCuratorFramework = AsyncCuratorFramework.wrap(curator)
```

Comment:
shouldn't this guy be private? To not expose the bare curator without defaults... (right now it's used just in the scope of this class)